### PR TITLE
Graphical interfaces

### PR DESCRIPTION
**Pull request by Terae**
By right-clicking on blocks, the appropriate GUI interface opens.

I updated these blocks converting `TextWidget` by `ButtonWidget`:
* **Battery**: launch `gnome-power-statistics` by default
* **CPU**: launch `gnome-system-monitor` by default
* **Net**: launch `nmtui` by default. Actually, it works only by clicking on text but not on a graphic.

The `on_click` argument is an `Option<String>` so it is still possible to open nothing like this:
```toml
[[block]]
name = "my_block"
on_click = ""
```

### CODE REVIEW

**Comment by atheriel** [comment]
FWIW, I'm strongly opposed to setting defaults that expect the user to have GNOME applications installed. This is very likely to annoy non-GNOME users -- which, for an i3-related application, is most of them.

I also think we should link this to a related issue: #131. It's important to note that if we decide to go this route -- left click to open a related application -- it prevents us from adding alternative left-click behaviours for these blocks in the future. For this reason I'm not terribly supportive of the change. What do you think, @greshake?

(There are also a few blocks that have on-click behaviour already, for example the temperature block.)

**Comment by Terae** [comment]
I agree in the fast we shouldn't setting defaults to GNOME applications, but does they are default GUI application installed on almost all systems or not? The easiest should be to set by default `on_click = ""` and let the user to choose its own GUI app.

Some blocks are toggled by the left-click event that is why I used the right-click behaviour to open graphic instances, except the temperature block (which doesn't differenciate teh mouse event yet) I fount no block which use this event so it seems to be a good and non destructive improvement no?

**Comment by BrendanBall** [comment]
I would certainly be one of those non-GNOME users. And no @Terae gnome apps are not installed on most systems. I would think a minority of i3 users have gnome installed. I would definitely leave it blank by default.

File: src/blocks/battery.rs [hunk 1]
```diff
@@ -215,6 +229,30 @@ impl Block for Battery {
         vec![&self.output]
     }
 
+    fn click(&mut self, event: &I3BarEvent) -> Result<()> {
```
Comment (l4l) [inline src/blocks/battery.rs]: Seems like all of these functions are the same, so it mgiht be a better idea to move it out from impl (maybe to ButtonWidget?) with a signature like `fn handle_click(id: &String, on_click: &String, event: &I3BarEvent) -> Result<()>` and for the impl left just a call for `handle_click(&self.id, &self.on_click, event)`

**Comment by l4l** [comment]
Is there any activity there? I think that feature is kinda useful

**Comment by atheriel** [comment]
@l4l I don't think there is particularly resounding support for this feature. Partly this is due to the non-trivial implementation, since we don't have any abstraction levels for blocks in the configuration file (and so every change needs to be made for blocks individually). And partly, I think, this is because it doesn't seem very "i3-like" to click the bar instead of using an application launcher like `drun` or `rofi`. Of course, that's more a cultural argument than a technical one.

**Pull request closed without merging**